File: it/structures/csharp/enum-default/Test/EnumLongNames.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.Runtime.Serialization;

namespace Test {
    /// Variants with long names.
    [JsonConverter(typeof(StringEnumConverter))]
    public enum EnumLongNames {
        [EnumMember(Value = "FooBar")]
        FOO_BAR,
        [EnumMember(Value = "Baz")]
        BAZ
    }
}

File: it/structures/java/enum-default/test/EnumLongNames.java
package test;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Variants with long names.
 */
public enum EnumLongNames {
    FooBar("FooBar"),
    Baz("Baz");

    String value;

    EnumLongNames(final String value) {
        this.value = value;
    }

    @JsonCreator
    public static EnumLongNames fromValue(final String value) {
        for (final EnumLongNames v : values()) {
            if (v.value.equals(value)) {
                return v;
            }
        }

        throw new IllegalArgumentException("value");
    }

    @JsonValue
    public String toValue() {
        return this.value;
    }
}
